.ci/scripts/utils.sh

```diff
@@ -166,3 +162,60 @@ do_not_use_nightly_on_ci() {
     exit 1
   fi
 }
+
+
+parse_args() {
```

Comment:
this is reaching the complexity level when I start wondering if it's easier overall to switch from a shell script to python. (not blocking, out of scope for this change)